src/settings/toml/migrations.rs

```diff
@@ -0,0 +1,504 @@
+                        "Multiple migrations with tag {} were found in your wrangler.toml",
+                        tag
+                    ),
+                    (None, _) => failure::bail!(
+                        "The migration with tag {} was not found in your wrangler.toml.",
+                        tag
+                    ),
+                }
+            }
+        }
+    }
+
+    fn coalesce_migrations(
+        migrations: &[MigrationConfig],
+        old_tag: Option<&str>,
+    ) -> Result<ApiMigration, failure::Error> {
+        if migrations.is_empty() {
+            failure::bail!("cannot coalesce empty migrations");
+        }
+        log::info!(
+            "coalescing migrations {:#?} after {:?}",
+            migrations,
+            old_tag
+        );
+        let mut new_classes = migrations[0].migration.durable_objects.new_classes.to_vec();
+        let mut deleted_classes = migrations[0]
+            .migration
+            .durable_objects
+            .deleted_classes
+            .to_vec();
+        let mut renamed_classes = migrations[0]
+            .migration
+            .durable_objects
+            .renamed_classes
+            .to_vec();
+        let mut transferred_classes = migrations[0]
+            .migration
+            .durable_objects
+            .transferred_classes
+            .to_vec();
+
+        for mc in &migrations[1..] {
+            // Renames will modify existing new, rename, and transfer migrations if they are present
+            for rename in mc.migration.durable_objects.renamed_classes.iter() {
+                if let Some(i) = new_classes.iter().position(|c| c == &rename.from) {
+                    new_classes[i] = rename.to.clone();
+                } else if let Some(existing) = renamed_classes
+                    .iter_mut()
+                    .find(|existing| existing.to == rename.from)
+                {
+                    existing.to = rename.to.clone();
+                } else if let Some(existing) = transferred_classes
+                    .iter_mut()
+                    .find(|existing| existing.to == rename.from)
+                {
+                    existing.to = rename.to.clone();
+                } else {
+                    // rename does not modify an existing migration, add it to the final migration
+                    renamed_classes.push(rename.clone());
+                }
+            }
+
+            // deletes will remove existing new, rename, and transfer migrations if they are present
+            for delete in mc.migration.durable_objects.deleted_classes.iter() {
+                if let Some(i) = new_classes.iter().position(|c| c == delete) {
+                    new_classes.remove(i);
+                } else if let Some(i) = renamed_classes
+                    .iter_mut()
+                    .position(|existing| &existing.to == delete)
+                {
+                    renamed_classes.remove(i);
```

Comment:
We need to apply the delete migration to the `from` class at this point, right? Otherwise it may stick around when it's supposed to be deleted.

Reply:
oo, good point